bot/glossary_command.go

```diff
@@ -0,0 +1,148 @@
+package bot
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/quintilesims/slackbot/db"
+	"github.com/quintilesims/slackbot/models"
+	glob "github.com/ryanuber/go-glob"
+	"github.com/urfave/cli"
+)
+
+// NewGlossaryCommand returns a cli.Command that manages !glossary
+func NewGlossaryCommand(store db.Store, w io.Writer) cli.Command {
+	return cli.Command{
+		Name:  "!glossary",
+		Usage: "manage the glossary",
+		Subcommands: []cli.Command{
+			{
+				Name:      "define",
+				Usage:     "add or set an entry in the glossary",
+				ArgsUsage: "KEY DEFINITION",
+				Action:    newGlossaryDefineAction(store, w),
+			},
+			{
+				Name:      "rm",
+				Usage:     "remove an entry from the glossary",
+				ArgsUsage: "KEY",
+				Action:    newGlossaryRemoveAction(store, w),
+			},
+			{
+				Name:      "search",
+				Usage:     "search for entries in the glossary",
+				ArgsUsage: "GLOB",
+				Action:    newGlossarySearchAction(store, w),
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "count",
+						Value: 10,
+						Usage: "The maximum number of entries to display",
+					},
+				},
+			},
+		},
+	}
+}
+
+func newGlossaryDefineAction(store db.Store, w io.Writer) func(c *cli.Context) error {
+	return func(c *cli.Context) error {
+		key := c.Args().Get(0)
```

Comment:
I had this before and you had me change it? If I use arg do you want me to do `arg.First() and arg.Tail()`

Reply:
I kinda forget what the past conversations were, all I'm saying is that I've seen the implementation of `c.Args()`, and it actually does some work, so it would be most efficient to call it once.